### ISSUE TITLE
CalDAV only works one way

### ISSUE THREAD

**Issue by marcusboden** [marcoancona/MineTime#121]
Hey there,

I'm trying to sync a CalDAV Account but I can't quite get it to work: No events are visible in Minetime. But if I create an event in Minetime, it will be displayed AND synced to the server.

here is what happens when I try to sync:
```
2019-05-27T08:33:04.197Z - error: CalDAV: error while running syncCalendar -  message=Bad status 404: <?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre_DAV_Exception_NotFound</s:exception>
  <s:message>Calendar object not found</s:message>
</d:error>
, stack=Error: Bad status 404: <?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre_DAV_Exception_NotFound</s:exception>
  <s:message>Calendar object not found</s:message>
</d:error>

    at Request._callback (/opt/MineTime/resources/app.asar/node_modules/dav/build/XMLHttpRequestWrapper.js:84:20)
    at Request.self.callback (/opt/MineTime/resources/app.asar/node_modules/dav/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/opt/MineTime/resources/app.asar/node_modules/dav/node_modules/request/request.js:1161:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/opt/MineTime/resources/app.asar/node_modules/dav/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1056:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9), code=404
2019-05-27T08:33:04.198Z - error: Error on sync of calendar 5ce7a6bc4ff12e708571f225. Next trial in in 30 secs... message=Bad status 404: <?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre_DAV_Exception_NotFound</s:exception>
  <s:message>Calendar object not found</s:message>
</d:error>
, stack=Error: Bad status 404: <?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre_DAV_Exception_NotFound</s:exception>
  <s:message>Calendar object not found</s:message>
</d:error>

    at Request._callback (/opt/MineTime/resources/app.asar/node_modules/dav/build/XMLHttpRequestWrapper.js:84:20)
    at Request.self.callback (/opt/MineTime/resources/app.asar/node_modules/dav/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/opt/MineTime/resources/app.asar/node_modules/dav/node_modules/request/request.js:1161:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/opt/MineTime/resources/app.asar/node_modules/dav/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1056:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9), code=404, timestamp=2019-05-27T08:33:04.197Z
```

The caldav addresses (and the password) should be correct, since they are working with vdirsyncer and with davdroid.

**Comment by marcoancona** [comment]
Several fixes to CalDAV have been implemented. This should be solved now. If it is not the case, feel free to comment or open another issue.

**Comment by marcusboden** [comment]
Still the same error, using the 1.8.4 release:

```error: Error on sync of calendar 5f17dea513fc6952c640ad30. Next trial in in 30 secs... message=Bad status 404: <?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre_DAV_Exception_NotFound</s:exception>
  <s:message>Calendar object not found</s:message>
</d:error>
, stack=Error: Bad status 404: <?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre_DAV_Exception_NotFound</s:exception>
  <s:message>Calendar object not found</s:message>
</d:error>

    at Request._callback (/tmp/.mount_MineTiDULJrs/resources/app.asar/node_modules/dav/dav.js:88:26)
    at Request.self.callback (/tmp/.mount_MineTiDULJrs/resources/app.asar/node_modules/dav/dav.js:29815:22)
    at Request.emit (events.js:210:5)
    at Request.<anonymous> (/tmp/.mount_MineTiDULJrs/resources/app.asar/node_modules/dav/dav.js:30791:10)
    at Request.emit (events.js:210:5)
    at IncomingMessage.<anonymous> (/tmp/.mount_MineTiDULJrs/resources/app.asar/node_modules/dav/dav.js:30713:12)
    at Object.onceWrapper (events.js:299:28)
    at IncomingMessage.emit (events.js:215:7)
    at endReadableNT (_stream_readable.js:1183:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21), code=404
```

**Comment by marcoancona** [comment]
Would it be possible to create a test account for me? You can contact me directly at connect —at— minetime.ai

**Comment by BlueSpacePotato** [comment]
Got the same error. 

```
2020-08-03T07:11:10.853Z - error:  message=Bad status 404: , stack=Error: Bad status 404: 
    at Request._callback (/opt/MineTime/resources/app.asar/node_modules/dav/dav.js:88:26)
    at Request.self.callback (/opt/MineTime/resources/app.asar/node_modules/dav/dav.js:29815:22)
    at Request.emit (events.js:210:5)
    at Request.<anonymous> (/opt/MineTime/resources/app.asar/node_modules/dav/dav.js:30791:10)
    at Request.emit (events.js:210:5)
    at IncomingMessage.<anonymous> (/opt/MineTime/resources/app.asar/node_modules/dav/dav.js:30713:12)
    at Object.onceWrapper (events.js:299:28)
    at IncomingMessage.emit (events.js:215:7)
    at endReadableNT (_stream_readable.js:1183:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21), code=404
```

**Comment by marcoancona** [comment]
Fixed in 1.8.6
